melbourne-school-of-theology.csl

```diff
@@ -496,24 +574,32 @@
       </if>
       <else-if variable="publisher-place publisher" match="any">
         <group delimiter="; " prefix=" (" suffix=")">
+          <group delimiter="; ">
+            <text macro="container-contributors-bibliography"/>
+            <text macro="locators-note"/>
+            <text macro="collection-title"/>
+          </group>
           <group>
-            <text term="volume" form="short" suffix=" "/>
-            <number variable="volume"/>
             <choose>
-              <if match="any" variable="number-of-volumes">
-                <text value="of" prefix=" " suffix=" "/>
+              <if match="all" variable="volume">
+                <text term="volume" form="short" suffix=" "/>
+                <number variable="volume"/>
+                <choose>
+                  <if match="any" variable="number-of-volumes">
+                    <text value="of" prefix=" " suffix=" "/>
+                  </if>
+                </choose>
               </if>
             </choose>
             <group>
               <number variable="number-of-volumes" form="numeric"/>
-              <text term="volume" form="short" plural="true" strip-periods="true" prefix=" "/>
+              <choose>
+                <if match="none" variable="volume">
```

Comment:
Yes, this is correct.  If there is a volume number, it is rendered "vol x of x".  If there is no volume number, because it is explicitly stated in the title, for example, but there is a number of volumes, then it is rendered "x vols".  It removes the redundancy of having whitespace in the form of "vol  of x" This is addressing a very specific, but very common case that is used in our university a lot.